pkg/ebpf/c/tracee.bpf.c

```diff
@@ -656,6 +664,382 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
+{
+    return bpf_map_lookup_elem(&unhidden_modules_map, &mod) ==
+           NULL; // if we don't find the module in our map, it means the module is hidden
+}
+
+static __always_inline bool find_modules_from_module_kset_list(program_data_t *p)
+{
+    char module_kset_sym[12] = "module_kset";
+    struct kset *mod_kset = (struct kset *) get_symbol_addr(module_kset_sym);
+    struct list_head *head = &(mod_kset->list);
+    struct kobject *pos = list_first_entry_ebpf(head, typeof(*pos), entry);
+    struct kobject *n = list_next_entry_ebpf(pos, entry);
+    bool finished_iterating = false;
+
+#pragma unroll
+    for (int i = 0; i < MAX_NUM_MODS; i++) {
+        if (&pos->entry == head) {
+            finished_iterating = true;
+            break;
+        }
+
+        if (READ_KERN(n->name) == NULL) { // Without this the list seems infinite. Also, using pos
+                                          // here seems incorrect as it starts from a weird member
+            finished_iterating = true;
+            break;
+        }
+
+        struct module_kobject *mod_kobj =
+            (struct module_kobject *) container_of(n, struct module_kobject, kobj);
+        if (mod_kobj) {
+            struct module *mod = READ_KERN(mod_kobj->mod);
+            if (mod) {
+                if (is_hidden((u64) mod)) {
+                    send_hidden_module((u64) mod, mod->name, p);
+                }
+            }
+        }
+
+        pos = n;
+        n = list_next_entry_ebpf(n, entry);
+    }
+
+    return !finished_iterating;
+}
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 20, 0)
+BPF_QUEUE(walk_mod_tree_queue, rb_node_t, 2048); // used to walk a rb tree
+
+static __always_inline bool walk_mod_tree(program_data_t *p, struct rb_node *root, int idx)
+{
+    struct latch_tree_node *ltn;
+    struct module *mod;
+    bool finished_iterating = false;
+    struct rb_node *curr = root;
+
+    #pragma unroll
+    for (int i = 0; i < MAX_NUM_MODS; i++) {
+        if (curr != NULL) {
+            rb_node_t rb_nod = {.node = curr};
+            bpf_map_push_elem(&walk_mod_tree_queue, &rb_nod, BPF_EXIST);
+
+            curr = READ_KERN(curr->rb_left); // Move left
+        } else {
+            rb_node_t rb_nod;
+            if (bpf_map_pop_elem(&walk_mod_tree_queue, &rb_nod) != 0) {
+                finished_iterating = true;
+                break;
+            } else {
+                curr = rb_nod.node;
+                ltn = __lt_from_rb(curr, idx);
+                mod = READ_KERN(container_of(ltn, struct mod_tree_node, node)->mod);
+
+                if (is_hidden((u64) mod)) {
+                    send_hidden_module((u64) mod, mod->name, p);
+                }
+
+                /* We have visited the node and its left subtree.
+                Now, it's right subtree's turn */
+                curr = READ_KERN(curr->rb_right);
+            }
+        }
+    }
+
+    return !finished_iterating;
+}
+
+    #ifdef CORE
```

Comment:
why only in CORE? what happens if you are not in CORE and kernel version higher the 4.20?